third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -2113,22 +2123,38 @@ func cleanFailedContainerCluster(d *schema.ResourceData, meta interface{}) error
 	return nil
 }
 
-func waitForContainerClusterReady(config *Config, project, location, clusterName string, timeout time.Duration) error {
-	return resource.Retry(timeout, func() *resource.RetryError {
+var containerClusterRestingStates = RestingStates{
+	"RUNNING":  ReadyState,
+	"DEGRADED": ErrorState,
+	"ERROR":    ErrorState,
+}
+
+func containerClusterAwaitRestingState(config *Config, project, location, clusterName string, timeout time.Duration) (state string, stateType StateType, err error) {
+	err = resource.Retry(timeout, func() *resource.RetryError {
 		name := containerClusterFullName(project, location, clusterName)
 		cluster, err := config.clientContainerBeta.Projects.Locations.Clusters.Get(name).Do()
 		if err != nil {
 			return resource.NonRetryableError(err)
 		}
-		if cluster.Status == "PROVISIONING" || cluster.Status == "RECONCILING" || cluster.Status == "STOPPING" {
-			return resource.RetryableError(fmt.Errorf("Cluster %q has status %q with message %q", clusterName, cluster.Status, cluster.StatusMessage))
-		} else if cluster.Status == "RUNNING" {
-			log.Printf("Cluster %q has status 'RUNNING'.", clusterName)
+
+		state = cluster.Status
+		stateType = containerClusterRestingStates[cluster.Status]
+
+		if stateType == ReadyState {
+			log.Printf("Cluster %q has status %q.", clusterName, state)
 			return nil
+		} else if stateType == ErrorState {
+			return resource.NonRetryableError(fmt.Errorf("Cluster %q has terminal state %q with message %q.", clusterName, state, cluster.StatusMessage))
 		} else {
-			return resource.NonRetryableError(fmt.Errorf("Cluster %q has terminal state %q with message %q.", clusterName, cluster.Status, cluster.StatusMessage))
+			return resource.RetryableError(fmt.Errorf("Cluster %q has status %q with message %q", clusterName, state, cluster.StatusMessage))
 		}
 	})
+
+	if _, ok := containerClusterRestingStates[state]; ok {
+		return state, stateType, nil
+	} else {
+		return "", UndefinedState, err
```

Comment:
you can simplify to 1 return here as `containerClusterRestingStates["garbage"] == UndefinedState`